sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1064,6 +1066,31 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
     check(plan)
   }
 
+  /**
+   * This method is only used for checking collected metrics. This method tries to
+   * remove extra alias-only project from the plan so that we can identify exact
+   * duplicates metric definition.
+   */
+  private def simplifyPlanForCollectedMetrics(plan: LogicalPlan): LogicalPlan = {
+    plan.resolveOperatorsDown {
+      case p: Project if p.projectList.size == p.child.output.size =>
+        val assignExprIdOnly = p.projectList.zip(p.child.output).forall {
+          case (left: Alias, right: Attribute) =>
+            if (left.child.semanticEquals(right) && right.name == left.name) {
```

Comment:
well, we can just write `left.child.semanticEquals(right) && right.name == left.name`, no if-else needed.

Reply:
good point